src/parser/jobs/rpr/modules/ImmortalSacrifice.tsx

```diff
@@ -0,0 +1,154 @@
+
+	static override handle = 'immortalsacrifice'
+	static override title = t('rpr.immortal-sacrifice.title')`Immortal Sacrifice stacks consumed`
+	static override debug = false
+
+	@dependency private brokenLog!: BrokenLog
+	@dependency private data!: Data
+	@dependency private checklist!: Checklist
+
+	private windows: ImmortalSacrificeWindow[] = []
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeStack)
+		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeExpiration)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.PLENTIFUL_HARVEST.id), this.onPlentifulHarvest)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onImmortalSacrificeStack(event: Events['statusApply']) {
+		let window = this.getOpenWindow()
+
+		if (window === null) {
+			window = {
+				startTime: event.timestamp,
+				closed: false,
+				consumed: false,
+				stackCount: 0,
+			}
+
+			this.windows.push(window)
+		}
+
+		window.stackCount = event.data ?? window.stackCount + 1
+	}
+
+	private onPlentifulHarvest(event: Events['action']) {
+		const window = this.getOpenWindow()
+
+		if (window === null) {
+			this.brokenLog.trigger(this, 'PH without immortal sacrifice stack',
+				<Trans id="rpr.immortal-sacrifice.trigger.no-immortal-sacrifice">
+					<DataLink action="PLENTIFUL_HARVEST"/> was executed but there were
+					no <DataLink status="IMMORTAL_SACRIFICE"/> stacks.
+				</Trans>)
+			return
+		}
+
+		this.debug('Closing window because plentiful harvest were used', window)
+
+		window.closed = true
+		window.endTime = event.timestamp
+		window.consumed = true
+	}
+
+	private onImmortalSacrificeExpiration(event: Events['statusRemove']) {
+		const window = this.getOpenWindow()
+
+		// valid case, means it was consumed
+		if (window === null) {
+			return
+		}
+
+		this.debug('Closing window that was expired before consumption', window)
+
+		window.closed = true
+		window.consumed = false
+		window.endTime = event.timestamp
+	}
+
+	private getOpenWindow() {
+		if (this.windows.length > 0) {
+			const w = this.windows[this.windows.length - 1]
+
+			if (!w.closed) {
+				return w
+			}
+
+		}
+
+		return null
+	}
+
+	private onComplete() {
+		this.debug(() => {
+			this.debug('Immortal sacrifice windows')
+
+			this.windows.forEach(w => {
+				const start = this.parser.formatEpochTimestamp(w.startTime)
+				const end = w.endTime != null ? this.parser.formatEpochTimestamp(w.endTime) : '<Not Closed>'
+
+				this.debug(`Start: ${start}, End: ${end}, Closed: ${w.closed}, Consumed: ${w.consumed}, Stack Count: ${w.stackCount}`)
+			})
+		})
+
+		const totalImmortalSacrifices = this.windows.length
+
+		if (totalImmortalSacrifices <= 0) {
+			return
+		}
```

Comment:
Why? If they got zero uses, that's a problem.

Reply:
Since it is tied to the use of the buff I thought this message wouldn't make much sense, I'll add another message in the case it's 0 telling you should use it.

Reply:
Maybe consider a Suggestion for that.

Reply:
Poor choice on my words, by message I meant a suggestion, added it.